techSpec.js

```diff
@@ -0,0 +1,145 @@
+/*
+2. Introduction
+
+a. Overview, Problem Description, Summary, or Abstract
+
+    Context: Cat API is an Express.js application run on Node.js runtime. 
+
+    Problem: The Cat API currently does not have access to a cat database that persists. 
+    The information is generated when the API server is started. 
+
+    Suggested solution: Implement a SQLite database which allows information to be persisted. 
+
+b. Glossary  or Terminology
+
+    parameterization - parameterized queries is a way of preventing SQL injection. This can be done using named placeholders. 
+    SQL - structured query language 
+    SQLite - a file-based database engine that uses SQL syntax
+    sqlite3 - node module enables connection to SQLite databases stored in memory or in a file 
+
+    sqlite3 methods:
+        db.all - allows us to access all the rows in the dataset - this might offer the best flexibility for now
+        db.each - executes code for each row in a dataset
+        db.get - executes code for the FIRST row - use when we know only one row will return. 
+        db.run - used to create / alter tables and to insert / update table data. 
+
+d. Goals or Product and Technical Requirements
+
+    Product requirements: 
+        1. We should have a pre-populated SQLite database on the file system. 
+        2. Upon start-up of the Cat API Express server, it should establish a connection to the SQL database - or should it?  
+        3. The Cat API should be able to read from the SQLite database for GET all & GET by id requests and return objects. 
+        4. The Cat API should be able to write to the SQLite database for POST & PUT requests and return new/updated objects. 
+        4. The Cat API should be able to delete from the SQLite database for DEL requests. 
+        
+
+    Technical requirements: 
+        
+        1.
+        Create and populate 'cat' & 'pedigree breed' tables via sqlite in Terminal. 'CREATE TABLE' and so forth....
+        Remove list of cats and breed objects from catsRepo.js & breedsRepo.js respectively.   
+        Review if the repositories still need to be classes in this case? 
+        Note that 'id' key in tables is to have PRIMARY KEY constraint (needs to be unique)  
+         
+        2. 
+        Install sqlite3 node module into the root directory of cat api. Import/require sqlite3 into repository files for use.
+
+        We will create a DAO (Data Access Object) which contains a class AppDAO that...
```

Comment:
Nice :) this is really great research and it's good that you're thinking about abstractions like this.